meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1634 @@
+
+        if (SMPL_version == 'SMPLX'):
+            # Use 300 shape model by default if available
+            model_file = SMPLX_MODELFILE
+
+        elif (SMPL_version == 'SUPR'):
+            model_file = SUPR_MODELFILE
+
+        # for now, the SMPLH option has been removed from the properties because we don't have regressors for it, 
+        # so afm and a bunch of other stuff doesn't work
+        elif (SMPL_version == "SMPLH"):
+            model_file = SMPLH_MODELFILE
+
+        else:
+            model_file = "error bad SMPL_version"
+
+        objects_path = os.path.join(PATH, "data", model_file, "Object")
+        object_name = SMPL_version + "-mesh-" + gender
+
+        bpy.ops.wm.append(filename=object_name, directory=str(objects_path))
+
+        # Select imported mesh
+        object_name = context.selected_objects[0].name
+        bpy.ops.object.select_all(action='DESELECT')
+        context.view_layer.objects.active = bpy.data.objects[object_name]
+        bpy.data.objects[object_name].select_set(True)
+
+        #define custom properties on the avatar itself to store this kind of data so we can use it whenever we need to
+        bpy.context.object['gender'] = gender
+        bpy.context.object['SMPL_version'] = SMPL_version
+
+        # add a texture and change the texture option based on the gender
+        # male texture if it's a male, female texture if it's female or neutral
+        if (gender == 'male'):
+            context.window_manager.smpl_tool.texture = "m"
+        else:
+            context.window_manager.smpl_tool.texture = "f"
+
+        bpy.ops.object.set_texture()
+        bpy.ops.object.reset_body_shape('EXEC_DEFAULT')
+
+        return {'FINISHED'}
+
+
+class OP_SetTexture(bpy.types.Operator):
+    bl_idname = "object.set_texture"
+    bl_label = "Set"
+    bl_description = ("Set selected texture")
+    bl_options = {'REGISTER', 'UNDO'}
+
+    @classmethod
+    def poll(cls, context):
+        try:
+            # Enable button only if in active object is mesh
+            if (context.object.type == 'MESH'):
+                return True
+            else:
+                return False
+        except: return False
+
+    def execute(self, context):
+        selection = context.window_manager.smpl_tool.texture
+        obj = bpy.context.object
+
+        # if there's no material, add one
+        if (len(obj.data.materials) == 0):
+            # the incoming name of the selected object is in the format of "SUPR-mesh-male"
+            # this line turns "SUPR-mesh-male" into "SUPR-male", which is the naming format of the materials
+            split_name = obj.name.split("-")
+            new_material_name = split_name[0] + "-" + split_name[2]
+            material = bpy.data.materials.new(name=new_material_name)
+            bpy.context.object.data.materials.append(material)
+
+        else: 
+            material = obj.data.materials[0]
+
+        # Enable the use of nodes for the material
```

Comment:
Minor philosophical aside, comments should be for explaining why rather than what you are doing. E.g. "enable the use of nodes" doesn't tell me any more than `material.use_nodes = True` does, but I _would_ like to know what "using nodes" actually means and why we need to turn them on.